dataline-server/src/main/java/io/dataline/server/ServerApp.java

```diff
@@ -115,6 +131,32 @@ public void configure() {
     server.join();
   }
 
+  private void setCustomerIdIfNotSet(final Path configRoot) {
```

Comment:
interesting thought. if we _only_ ever want to use customerId in the tracker, then i'd be fine with that. in the current iteration of the PR, i've added `customerId` as part of the API interface in the WorkspaceRead struct. If we think `customerId` is useful outside of just tracking and should stay in the API interface, then I think it's better for it to be in `ServerApp` than for it to be in `TrackingClientSingleton`, because then the `ServerApp` and its data model would have an implicit dependency `TrackingClientSingleton` being initialized. If we want to confine customerId just to tracking, then yeah, happy to stick this in `TrackingClientSingleton`.

Reply:
Ok let's keep that as a hack then.